src/streamsync/core.py

```diff
@@ -879,6 +885,13 @@ def __init__(self, session_id: str, cookies: Optional[Dict[str, str]], headers:
         new_state.user_state.mutated = set()
         self.session_state = new_state
         self.event_handler = EventHandler(self)
+        self.component_manager = ComponentManager()
```

Comment:
I am not convince about the pattern integrating a manager in an instanced class.

I would advocate to keep the list of components the ComponentManager but we would ingest them with the session id. The component manager would stay a high level class we can rely on.

I would make the signature of ComponentManager evolve to keep the list of component per session id. It would be more easier to inspect application memory model and to make it evolve. We know all the components from all the session are in the ComponentManager. It may also allow advanced optimization as storing the diff between static component declaration and dynamic declaration.

We would also keep the ability to query the ComponentManager application wide. It may be interesting for some pattern.

Reply:
I think it's aligned with the current architecture. We're already managing sessions for State, my idea is to just put components there so we can manage them in the same way. There's a mechanism for example to clear out expired sessions and we'd have to replicate it for components.